osu.Game/Skinning/SkinnableTargetComponent.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Game.Skinning
+{
+    public class SkinnableTargetComponent : ISkinComponent
```

Comment:
```
ISkinnableComponent
ISkinComponent
ISkinSerialisable
ISkinnableTarget

SkinnableTargetComponent : ISkinComponent (doesn't inherit either ISkinnableComponent or ISkinnableTarget??)
SkinnableTarget (doesn't inherit ISkinnableTarget)
```
This is all very hard to follow because naming is a mess.

I'm not sure on the name of `SkinnableTargetWrapper` and `SkinnableElementTargetWrapper` either. Like there's one-too-many `Target`s here.

Reply:
The last two are definitely pain points. I was following existing naming but agree they don't make things any clearer. The first is a "lookup" type. Should we renaming that whole hierarchy to SkinLookup instead of SkinComponent?

We could also choose a synonym for `Target` like `Area` or something.

Reply:
Since `SkinnableTargetWrapper` is for components of a given target, how about `SkinnableComponentsContainer` instead?

Reply:
> The first is a "lookup" type

Funny, because it's actually the _second_ that is the lookup type :stuck_out_tongue:

But I mean, `ISkinComponent` is more what I'd expect for a lookup type than I'd expect `ISkinnableComponent` to be a drawable type. Almost feels like `ISkinnableComponent` should be called `ISkinnable`.

Reply:
I've applied a bit of renaming, see how things feel now.

Reply:
Much better. There's still a bit of weirdness regarding `SkinnableDrawable` that I forgot was a thing, but it's more digestible now. Also `SkinnableTargetWrapper` is a bit weird now that there's a `SkinnableTargetContainer` - perhaps a better name is `SkinnableTargetElementContainer` or something like that.

Reply:
`SkinnableDrawable` should mostly be out of the equation with the new classes. It's still something we can keep around into the future, though.